app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt

```diff
@@ -116,8 +129,9 @@ class AwesomeBarView(
             sessionProvider =
                 SessionSuggestionProvider(
                     components.core.sessionManager,
-                    components.useCases.tabsUseCases.selectTab,
-                    components.core.icons
+                    selectTabUseCase,
+                    components.core.icons,
+                    excludeSelectedSession = true
```

Comment:
I'm a bit confused, as this is a fixed value (`true`), the selected session will never be shown and as result the function `onExistingSessionSelected` is never going to be executed.

If the above it's the expected behavior is not better to give an  empty implementation to  `selectTabUseCase`? Or for these case  in AC we could make this optional.

```kotlin
         val selectTabUseCase = object : TabsUseCases.SelectTabUseCase {
            override fun invoke(session: Session) =  Unit
        }
```

Reply:
@Amejia481 Excluding the currently selected session seems like a good default. No user would want a suggestion to switch to their current tab from an AwesomeBar search query. If that's not how this works, then I'm shocked. I did test the code to verify that it switches tabs as expected and does not include the current tab in results. It only shows other tabs.

`onInputChanged()` is adding the `Awesomebar.Suggestion` only if `shouldIncludeSelectedSession()` comes up true for the current session in `SessionSuggestionProvider`. `shouldIncludeSelectedSession()` only comes up false if the current session is not the selected session. In other words, this code should work as I just described above. If that's not how this works, then we really need to add some KDoc to clarify that these functions do not match their names.

Reply:
Sorry for the misunderstanding, I double checked and you are right :thumbsup: